### PR TITLE
Integrate prettier, setup precommit hooks, format all files

### PR DESCRIPTION
**Pull request by anshumanv**
* Integrated prettier with ESLint
* Formatted all files
* Added precommit hooks to format files before commit.

@montogeek @EugeneHlushko Looking forward to your feedback! :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,8 +57,10 @@
     }
   },
   "lint-staged": {
-    "*.{js,jsx,md}": [
-      "npm run lint:js"
+    "*.{js,jsx,md,scss}": [
```
Comment (montogeek) [inline package.json]: Given that you created scripts for each type of file, please separate this as well, editing a SCSS file then run prettier for JS files is a waste of time

File: package.json [hunk 2]
```diff
@@ -83,6 +87,7 @@
     "directory-tree-webpack-plugin": "0.3",
     "duplexer": "^0.1.1",
     "eslint": "4.19.1",
+    "eslint-config-prettier": "^4.2.0",
```
Comment (montogeek) [inline package.json]: Please pin exact version as you did with `prettier`

**Comment by EugeneHlushko** [comment]
From the top of my head:
- change jsx element closing tag to not be on the newline for multiline cases
- allow trailing comma (we can enfroce as well)
- change the order of calling the format tasks, format -> lint -> throw if something wasnt fixed automatically

**Comment by montogeek** [comment]
> change jsx element closing tag to not be on the newline for multiline cases

I prefer it in a new line, thanks James for prettier :D

File: package.json (hunk 2 above)
Comment (anshumanv) [reply package.json]: Pinned

File: package.json (hunk 1 above)
Comment (anshumanv) [reply package.json]: Ah missed this, thanks!

**Comment by anshumanv** [comment]
Enforced trailing comma, updated order and keeping multiline cased in new line

**Comment by anshumanv** [comment]
@montogeek @EugeneHlushko formatting md with prettier seems to use emphasis style as `*` but markdownlink used `-` I can't seem to find a way to enforce md style via prettier, how about we switch to `*` as emphasis style or are there some tradeoffs which I'm missing?

**Comment by anshumanv** [comment]
You can try running `npm run format:markdown` for a quick check

**Comment by chenxsan** [comment]
Closed in favor of https://github.com/webpack/webpack.js.org/pull/4518. Thanks anyway.

**Comment by anshumanv** [comment]
At least this got closed now after 2 years, thanks for picking it up!

**Pull request closed without merging**